install/kubernetes/cilium/templates/cilium-agent/service.yaml

```diff
@@ -44,3 +45,28 @@ spec:
     targetPort: envoy-metrics
 {{- end }}
 {{- end }}
+{{- if and .Values.hubble.enabled .Values.hubble.listenAddress .Values.hubble.peerService.enabled }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: hubble-peer
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: cilium
+spec:
+  selector:
+    k8s-app: cilium
+  ports:
+  - name: peer-service
+    {{- if .Values.hubble.relay.servicePort }}
+    port: {{ .Values.hubble.peerService.servicePort }}
+    {{- else }}
+    port: {{ .Values.hubble.tls.enabled | ternary 443 80 }}
+    {{- end }}
+    protocol: TCP
+    targetPort: {{ splitList ":" .Values.hubble.listenAddress | last }}
+{{- if semverCompare ">=1.22-0" .Capabilities.KubeVersion.GitVersion }}
+  internalTrafficPolicy: Local
```

Comment:
TIL about `internalTrafficPolicy`, thanks @nathanjsweet!

From the [documentation](https://kubernetes.io/docs/concepts/services-networking/service/#internal-traffic-policy):
> If the traffic policy is `Local` and there are no node-local endpoints, traffic is dropped by kube-proxy.

I understand this is mapping the current local unix domain socket behavior where Relay use the peer service from the co-located cilium agent, so that make some sense. However, I'm not sure I graps why we want to restrict this over TCP. Thoughts? cc @gandro @rolinh

Reply:
Maybe as a side note (I think raised `internalTrafficPolicy` as an option at some point): Cilium as a CNI doesn't support `internalTrafficPolicy` yet. So this right now is a no-op. But Cilium will likely support it at some point, so I think it's fine to keep (tracking issue https://github.com/cilium/cilium/issues/16071)

The only other way I could think of is to force the local backend would to access the TCP port via the local NodeIP, e.g. as follows (and then have Relay connect to the peer service at `os.GetEnv("PEER_SERVICE_IP")`:

```yaml
        env:
        - name: PEER_SERVICE_IP
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
```

But my understanding is that an explicit service is desired to make the system a bit more understandable.

> However, I'm not sure I graps why we want to restrict this over TCP.

I'm not sure I understand this question. The mTLS API is exposed via TCP after all? We want to avoid the Unix domain socket due SELinux policies, where as TCP services are less restricted by SELinux.